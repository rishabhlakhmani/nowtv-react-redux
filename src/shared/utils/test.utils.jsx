import React from 'react'
import { render } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import membersReducer from '../../store/members/members.slice'
import messagesReducer from '../../store/members/members.slice'

export function renderWithProviders(
  ui,
  {
    preloadedState = {
        messages: {
            messages: []
        },
        members: {
            members: []
        },
    },
    store = configureStore({
      reducer: { messages: messagesReducer, members: membersReducer },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}
