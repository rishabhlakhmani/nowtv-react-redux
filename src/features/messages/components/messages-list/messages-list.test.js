import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { screen } from '@testing-library/react';
import MessagesList from './messages-list.component';
import { renderWithProviders } from '../../../../shared/utils/test.utils';

const MESSAGES = [
  {
    id: '0b971ce3-85de-42a1-8b36-6d5ffd6df57c',
    userId: '2984bb94-d4ad-4f51-8885-757e42b19fc9',
    message:
      'In hac habitasse platea dictumst. Morbi vestibulum, velit id pretium iaculis, diam erat fermentum justo, nec condimentum neque sapien placerat ante.',
    timestamp: '2016-09-15T09:28:18Z',
  },

  {
    id: '1469e146-075f-46d2-888b-8d3152da1be9',
    userId: 'cff77726-363e-4ba6-a953-4dc431cdd383',
    message: 'Morbi non lectus. Aliquam sit amet diam in magna bibendum imperdiet.',
    timestamp: '2016-08-21T02:55:32Z',
  },
];

describe('MessagesList', function () {
  it('should render the component', function () {
    let messagesList = renderWithProviders(
      <MemoryRouter>
        <MessagesList messages={MESSAGES} />
      </MemoryRouter>
    );
    expect(messagesList.container).toMatchSnapshot();
  });

  it('should render the Messages List', function () {
    renderWithProviders(
      <MemoryRouter>
        <MessagesList messages={MESSAGES} />
      </MemoryRouter>
    );

    expect(screen.getAllByTestId('message-card').length).toBe(2);
  });
});
