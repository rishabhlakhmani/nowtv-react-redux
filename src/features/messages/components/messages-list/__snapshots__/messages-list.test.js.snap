// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList should render the component 1`] = `
<div>
  <div
    class="w-100 d-flex flex-wrap justify-content-center messages-container"
  >
    <div
      class="card-container card"
      data-testid="message-card"
    >
      <img
        alt="Image"
        class="card-img-top"
        src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
           
        </div>
        <div
          class="message"
        >
          <p
            class="card-text"
          >
            In hac habitasse platea dictumst. Morbi vestibulum, velit id pretium iaculis, diam erat fermentum justo, nec condimentum neque sapien placerat ante.
          </p>
          <span
            class="css-tooltip custom-tooltip"
          />
        </div>
      </div>
      <div
        class="card-footer"
      >
        <small
          class="text-muted"
        >
          9/15/2016, 10:28:18 AM
        </small>
      </div>
    </div>
    <div
      class="card-container card"
      data-testid="message-card"
    >
      <img
        alt="Image"
        class="card-img-top"
        src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
           
        </div>
        <div
          class="message"
        >
          <p
            class="card-text"
          >
            Morbi non lectus. Aliquam sit amet diam in magna bibendum imperdiet.
          </p>
          <span
            class="css-tooltip custom-tooltip"
          />
        </div>
      </div>
      <div
        class="card-footer"
      >
        <small
          class="text-muted"
        >
          8/21/2016, 3:55:32 AM
        </small>
      </div>
    </div>
  </div>
</div>
`;
